awx/api/serializers.py

```diff
@@ -5487,9 +5487,17 @@ def get_field_from_model_or_attrs(fd):
             raise serializers.ValidationError("Can only create instances on Kubernetes or OpenShift.")
         node_type = get_field_from_model_or_attrs("node_type")
         peers_from_control_nodes = get_field_from_model_or_attrs("peers_from_control_nodes")
+        peers = get_field_from_model_or_attrs("peers")
 
         if peers_from_control_nodes and node_type not in (Instance.Types.EXECUTION, Instance.Types.HOP):
             raise serializers.ValidationError("peers_from_control_nodes can only be enabled for execution or hop nodes.")
+
+        if node_type in (Instance.Types.CONTROL) and len(peers) > 0:
+            if self.instance and set(self.instance.peers.all()) != set(attrs['peers']):
+                raise serializers.ValidationError(
+                    "Peers is only possible for execution or hop node types. Set 'peers' via 'peers_from_control_nodes' instead. This value cannot be modified once set."
```

Comment:
Setting peers manually for control nodes is not allowed. Enable peers_from_control_nodes on the hop and execution nodes instead.